src/main/java/co/cask/hydrator/salesforce/plugin/source/streaming/SalesforceStreamingSource.java

```diff
@@ -154,6 +158,51 @@ private StructuredRecord getStructuredRecord(String jsonMessage) throws Connecti
     }
   }
 
+  private Object convertValue(Object value, Schema.Field field) {
+    if (value == null) {
+      return null;
+    }
+
+    Schema fieldSchema = field.getSchema();
+
+    if (fieldSchema.isNullable()) {
+      fieldSchema = fieldSchema.getNonNullable();
+    }
+
+    Schema.Type fieldSchemaType = fieldSchema.getType();
+    Schema.LogicalType logicalType = fieldSchema.getLogicalType();
+
+    if (fieldSchema.getLogicalType() != null) {
+      String valueString = (String) value;
+      switch (logicalType) {
+        case DATE:
+          return Math.toIntExact(ChronoUnit.DAYS.between(Instant.EPOCH, Instant.parse(valueString)));
```

Comment:
I see a lot of different time classes being used. Is it possible to use a class already used for this instead of bringing in a new one?

Reply:
Basically here he we are calculating how many days since 1970 passed to the given date. I find one method which does not require that other lib. But it requires timezone which we don't really want, since we don't know which timezone is that time from.